### PR TITLE
Add `:decimal` aggregate support to `DataLayer.Ets` [WIP]

### PR DESCRIPTION
**Pull request by NduatiK**
# Contributor checklist

- [x] Bug fixes include regression tests
- [ ] Chores
- [ ] Documentation changes
- [ ] Features include unit/acceptance tests
- [ ] Refactoring
- [ ] Update dependencies

### CODE REVIEW

File: lib/ash/data_layer/ets/ets.ex [hunk 1]
```diff
@@ -742,23 +742,37 @@ defmodule Ash.DataLayer.Ets do
...
+
+        first_item = List.first(Enum.to_list(Stream.take(items, 1)))
+
+        initial_value =
+          if is_struct(first_item, Decimal) do
+            Decimal.new(0)
           else
-            records
-            |> Stream.map(&Map.get(&1, field))
-            |> Stream.uniq()
```
Comment (NduatiK) [inline lib/ash/data_layer/ets/ets.ex:754]: This is unrelated to `decimal`, but I removed the `Stream.uniq` here when `uniq?` is false.

**Review by zachdaniel — APPROVED** [review]

**Comment by zachdaniel** [comment]
Looks like some dialyzer failures.

**Comment by NduatiK** [comment]
Looks like I need to improve my nil handling. Let me make some changes

**Comment by NduatiK** [comment]
All done!

**Review by zachdaniel — APPROVED** [review]

**Pull request merged**